app/helpers/api_page_helper.rb

```diff
@@ -0,0 +1,50 @@
+module ApiPageHelper
+  # ask for current_weather
+  def get_weather(city = 'berlin', api_id)
+    # Rails.cache.clear
+    if api_id == 1
+      cache_name = 'curr_weather'
+    elsif api_id == 2
+      cache_name = 'forecast_weather'
+    end
+
+    exists_weather_of(city, api_id, cache_name)
+    if api_id == 1
+      get_curr_weather_data(Rails.cache.fetch('curr_weather'))
+    elsif api_id == 2
+      get_forecast_weather_data(Rails.cache.fetch('forecast_weather'))
+    end
+  end
+
+  def exists_weather_of(city, api_id, cache_name)
+    if Rails.cache.fetch(cache_name)
+      test = Rails.cache.fetch(cache_name)
+      if test['name'] == city.capitalize
+      else
+        get_curr_or_forecast_weather(city, api_id)
```

Comment:
Looks like you're passing in the city in a few places in this helper. When you build a new object to encapsulate the API logic, make sure it gets instantiated with a city name, e.g. `NewApiObject.new("Berlin")`. Then with attr_reader or some method, you will have the city name available within the whole class.